### DISCUSSION TITLE
ngtcp2_conn_get_max_data_left/ngtcp2_conn_get_max_stream_data_left/ngtcp2_conn_get_send_quantum

### DISCUSSION THREAD

**Discussion by redbaron** [ngtcp2/ngtcp2#1409] [Q&A]
when creating packets for  multiple streams at once, how much data should be written? example client uses `ngtcp2_conn_get_send_quantum`  as an overall limit and rechecking `ngtcp2_conn_get_max_data_left` on every iteration. In my case I am preparing multiple packet buffers at once in a library and hand them over to app to be sent with `sendmmsg` , so there is no packets sent in between `ngtcp2_conn_writev_stream` calls. 

Why example choses  `ngtcp2_conn_get_send_quantum`  and not others to use as limit? when and how `ngtcp2_conn_get_max_stream_data_left` should be used, if at all, in this scenario?

**Comment by tatsuhiro-t** [comment]
The example client uses GSO and its limit is ngtcp2_conn_get_send_quantum.  That's why we solely used it.
ngtcp2_conn_get_max_stream_data_left is handled inside ngtcp2_conn_writev_stream, and app normally does not need to use it.
